### PR TITLE
refactor: keyboard shortcuts now supports different keyboard layouts including Dvorak

### LINKED ISSUE #3322: [bug]: CMD-C on Dvorak keyboard causes Reset Request
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

The "c" key in the Dvorak keyboard layout is in the position of the "i" key in the QWERTY layout. For some reason when I use a Dvorak layout and press CMD-C, it sends a CMD-I shortcut to Hoppscotch and resets the request. Other shortcuts work as expected.

Maybe this should be filed as a separate bug, but resetting a request cannot be undone, meaning that I lose all my work if I forget about this bug and try to copy some text. If it is difficult to implement undo for this action, I suggest this shortcut be removed since it's such a footgun.

### Steps to reproduce

1. Use a Dvorak layout (on Mac OS)
2. Select some text in the body of a request
3. Press CMD-C

 I've tested this in Firefox, Chrome, and the PWA, but not in other operating systems.

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by JoelJacobStephen**
### Ticket

- Closes HFE-219
- Fixes #3322
- Suppressed #3314

### Description
This PR focuses on allowing the keyboard shortcuts in the app to support different keyboard layouts by using the unique key code parameter. This has been tested with QWERTY and DVORAK Keyboard Layouts and it works as expected.

### Objectives
- [x] Keyboard shortcuts should support DVORAK layout 
- Example: `CMD/CTRL - C` used to reset the request when using DVORAK Layout which has now been resolved. Only `CMD/CTRL - I` resets the request connection on both QWERTY and DVORAK Layouts

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Review by liyasthomas — APPROVED** [review]
LGTM 💯

File: packages/hoppscotch-common/src/helpers/keybindings.ts [hunk 1]
```diff
@@ -132,17 +133,16 @@ function getPressedKey(ev: KeyboardEvent): Key | null {
   if (isLetter) return ev.code.toLowerCase().substring(3) as Key
 
   // Check if number keys
-  if (val.length === 1 && !isNaN(val as any)) return val as Key
-
-  // Check if question mark
-  if (val === "?") return "?"
+  const isDigit = ev.code.toLowerCase().startsWith("digit")
+  if (isDigit) return ev.code.toLowerCase().substring(5) as Key
```
Comment (jamesgeorge007) [inline packages/hoppscotch-common/src/helpers/keybindings.ts:137]:
```suggestion
  const isDigit = val.startsWith("digit")
  if (isDigit) return val.substring(5) as Key
```

File: packages/hoppscotch-common/src/helpers/keybindings.ts [hunk 2]
```diff
@@ -132,17 +133,16 @@ function getPressedKey(ev: KeyboardEvent): Key | null {
   if (isLetter) return ev.code.toLowerCase().substring(3) as Key
```
Comment (jamesgeorge007) [inline packages/hoppscotch-common/src/helpers/keybindings.ts:133]:
```suggestion
  if (isLetter) return val.substring(3) as Key
```

Comment (jamesgeorge007) [reply packages/hoppscotch-common/src/helpers/keybindings.ts:133]: Can't add the suggestion for the line above `132`.

**Review by jamesgeorge007 — COMMENTED** [review]
Can't we use the `val` variable initialized at the top at all places?

File: packages/hoppscotch-common/src/helpers/keybindings.ts (hunk 1 above)
Comment (JoelJacobStephen) [reply packages/hoppscotch-common/src/helpers/keybindings.ts]: @jamesgeorge007  I have made the changes.

**Review by AndrewBastin — APPROVED** [review]
lgtm 💯

**Pull request merged**